src/main/java/ro/msg/learning/shop/services/implementations/GenericServiceImpl.java

```diff
@@ -32,21 +33,21 @@ public Entity getById(Long id) {
         if (optionalEntity.isPresent()) {
             return optionalEntity.get();
         }
-        throw new NotFoundException("Entity not found!");
+        throw new NotFoundException(notFoundMessage);
     }
 
-    public void update(Long id, Entity newEntity) {
+    public Entity update(Long id, Entity newEntity) {
```

Comment:
the service gives back an entity which then is mapped in the controller to become a dto and travel to the front end. This is good, dont change the return type here :)